### PR TITLE
Enhance SumArray tests with edge cases and better assertions

### PR DESCRIPTION
**Pull request by pali101**
- Added tests for single elements, zeros, and large numbers
- Included random inputs for broader coverage

### CODE REVIEW

File: SumArray/test/SumArray.t.sol [hunk 1]
```diff
@@ -25,10 +26,87 @@ contract SumArrayTest is Test {
...
+        uint256[] memory arr = new uint256[](3);
+        arr[0] = 2 ** 128; // Large number
+        arr[1] = 2 ** 42; // Large number
+        arr[2] = 2 ** 200; // Large number
+        uint256 x = sumArray.sumArray(arr);
+        assertEq(
+            x,
+            (2 ** 128) + (2 ** 42) + (2 ** 200),
+            "expected sum of large numbers to match (2 ** 128) + (2 ** 42) + (2 ** 200)"
+        );
```
Comment (tanim0la) [inline SumArray/test/SumArray.t.sol:72]: i think throwing the actual number in the err message will be clearer
```
        assertEq(
            x,
            (2 ** 128) + (2 ** 42) + (2 ** 200),
            "expected sum of large numbers to match 1606938044258990275542302374708083540985666368394622650023936"
        );
```

Comment (pali101) [reply SumArray/test/SumArray.t.sol]: Thanks for the suggestion! I've updated the error message to include the actual sum value for clarity as recommended.

**Comment by tanim0la** [comment]
Thank you!
